File: Java/TP3_MET_JAVA/src/Inductif/RepresentationInductif.java
package Inductif;

public class RepresentationInductif implements VisiteurMotInductif<String> {

    @Override
    public String casVide() {
        return "";
    }

    @Override
    public String casCaractereMot(Character c, String m) {
        return c + m;
    }
}

File: TypeScript/TP3_MET_TS/src/Inductif/RepresentationInductif.ts
import {VisiteurMotInductif} from "./VisiteurMotInductif";

export class RepresentationInductif implements VisiteurMotInductif<string> {
    public casVide(): string {
        return "";
    }

    public casCaractereMot(c: string, m: string): string {
        return c + m;
    }
}
